clippy_lints/src/methods/option_map_unwrap_or.rs

```diff
@@ -74,16 +77,34 @@ pub(super) fn check<'tcx>(
             return;
         }
 
+        let mut suggest_is_some_and = false;
```

Comment:
nit: As this isn't mutated past the following `if`, let's make this something like
```rust
let suggest_is_some_and = if msrv.meets(msrvs::OPT_IS_SOME_AND) {
} else {
	false
}
```
(@xFrednet, just curious, if you know, can you suggest an easier way to do this? i.e., make a variable `true` if a let chain evaluates to `true`...)

Reply:
i reckon taking advantage of the `matches!` macro would work quite well as the only purpose of the `if let` blocks are pattern matching. 

something like this would work:
```
let suggest_is_some_and: bool = msrv.meets(msrvs::OPTION_IS_SOME_AND)
    && matches!(&unwrap_arg.kind, ExprKind::Lit(lit) 
    if matches!(lit.node, rustc_ast::LitKind::Bool(false)));
```

idk if its sacrificing readability at thought point though. imho, it's fine.

Reply:
Yeah that could be fine here though let chains with then `true` else `false` is pretty common, so having a better way would be nice

Reply:
> (@xFrednet, just curious, if you know, can you suggest an easier way to do this? i.e., make a variable true if a let chain evaluates to true...)

Nope, sadly not that I know of. It's possible to use the `if_chain` macro, but that would be less readable than your suggested alternative. I've also noticed that Rust is missing something for this. There was an RFC, to make `let <pat> = <expr>` evaluate to a boolean in every context, but that one got rejected IIRC.

A similar way to write your suggestion would also be:
```rs
let suggest_is_some_and;
if msrv.meets(msrvs::OPT_IS_SOME_AND) {
	suggest_is_some_and = true;
} else {
	suggest_is_some_and = false;
}
```
This can be more readable, in some circumstances, but often it's just a style question :)